### ISSUE TITLE
Discrepancy between hiding score overlay in osu!lazer and osu!stable

### ISSUE THREAD

**Issue by fluffetyfluff** [ppy/osu#12039]
osu!stable currently shows the hit error graph (and key overlay) even while the score overlay is disabled.
![image](https://user-images.githubusercontent.com/4528540/111539963-e423a800-8744-11eb-8039-ed2b78de4731.png)

osu!lazer, on the other hand, doesn't show anything. This might be because the option is named "HUD overlay visibility mode" (interestingly enough, the notification still reads "The score overlay is currently disabled"), but I do not think disabling the entire HUD is the point of disabling the score overlay. In my opinion, disabling score is to reduce distractions while playing, but the hit error bar is an integral part of gameplay.
![image](https://user-images.githubusercontent.com/4528540/111539724-960ea480-8744-11eb-88a4-2512b5f464d6.png)

Is this an intended change?

**Comment by bdach** [comment]
This might be eventually addressed by either: more skin/HUD customisation options or legacy layouts (https://github.com/ppy/osu/issues/4632), or alternate methods of displaying the error bar (https://github.com/ppy/osu/issues/3209).

I see it as a matter of personal preference. Personally I don't miss the hit error bar one iota (I like having the HUD fully off), but obviously if it is deemed that the general consensus is that it should unconditionally return for everyone, so be it.

**Comment by Joehuu** [comment]
Duplicate of https://github.com/ppy/osu/issues/11322, and that closed issue is broadly tracked in https://github.com/ppy/osu/issues/4924. @bdach not sure if I should close this as I believe specific issues should be kept open.

**Comment by fluffetyfluff** [comment]
Ah naturally I forgot to delete is:open when searching the issue tracker. Oops!
I think having a separate option to remove the hit error bar might work better (keeping Shift+Tab functionality similar to stable).

**Comment by bdach** [comment]
I suppose we can keep this open for searchability given it's been reported multiple times now...

**Comment by peppy** [comment]
The path forward here is likely going to involve adding support to the skin layout editor to let the user choose when certain elements appear. I'm intending to get to this but still figuring out the UX (how things should be exposed/named to provide enough functionality while still keeping it understandable).

Once this is done, the `Shift-Tab` cycle options would return to a `bool` selection, with the extra overloaded behaviour coming from the user's layout settings.

**Comment by ana-rchy** [comment]
bump for wanting more attention on this

**Comment by cesarila2** [comment]
Before coming to this issue, I was hoping that the "Playfield ("osu!" only)" would stay enabled when HUD is hidden. As it stands, I'm not sure what the difference between "HUD ("osu!" only)" and "Playfield ("osu!" only)" is supposed to be.

**Comment by bdach** [comment]
> what the difference between "HUD ("osu!" only)" and "Playfield ("osu!" only)" is supposed to be

The playfield layer allows you to attach things to the playfield rather than to corners of the screen.

**Comment by peppy** [comment]
> The playfield layer allows you to attach things to the playfield rather than to corners of the screen. It's not supposed to impact visibility in any way.

It might actually make sense for things on this layer to be visible always, given they are supposed to be attached to the playfield 🤔 .
